### PR TITLE
update proxy sha

### LINKED ISSUE #17374: Update Envoy to support new ALS protocol
I create this issue is to track update Envoy to support new ALS protocol added in https://github.com/envoyproxy/envoy-wasm/pull/205/commits/1f8c137605326f7c04e1eb1c1e503cae43360bfb

 - ~~Update Envoy in istio/proxy. https://github.com/istio/proxy/pull/2432~~
 - Update proxy in istio. #17403

### PR DESCRIPTION
**Pull request by hanahmily**
Signed-off-by: Hongtao Gao <hanahmily@gmail.com>

Please provide a description for what this PR is for.

Fixed #17374

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[x] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by hanahmily** [comment]
FYI @rshriram

**Comment by hzxuzhonghu** [comment]
@lizan what does this  `PRE_INITIALIZING` mean? I have seen it takes about more than 20s for the envoy to turn from `PRE_INITIALIZING` to `LIVE`.  And during this time, it does not connect to xds server either.

**Comment by lizan** [comment]
@hzxuzhonghu https://github.com/envoyproxy/envoy/issues/5862

**Review by rshriram — APPROVED** [review]

**Comment by hanahmily** [comment]
@rshriram all of the test passed.
For the envoy integration test, I remove `dynamic_resources` from template JSON because if there's no pilot in testing envoy will be stuck in `PRE_INITIALIZING` status.

and I rename `envoy.wasm` to `envoy.filters.http.wasm` according to https://github.com/istio/proxy/pull/2421/files

**Review by lizan — COMMENTED** [review]
/lgtm
/approve

FWIW, deleting xds config in instance test is legit. The test doesn't set up xDS server but testing with xDS config in bootstrap, now the case is no longer considered "LIVE" since https://github.com/envoyproxy/envoy/pull/8152.

**Review by lizan — APPROVED** [review]

**Comment by howardjohn** [comment]
thanks for the reference Lizan!

File: tests/integration/telemetry/stats/prometheus/testdata/metadata_exchange_filter.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ spec:
       patch:
         operation: INSERT_BEFORE
         value:
-          name: envoy.wasm
+          name: envoy.filters.http.wasm
```
Comment (howardjohn) [inline tests/integration/telemetry/stats/prometheus/testdata/metadata_exchange_filter.yaml]: I realize you are not making this change but it was made in istio/proxy at some time, but how is this supposed to work? how can a user upgrade ?

Comment (lizan) [reply tests/integration/telemetry/stats/prometheus/testdata/metadata_exchange_filter.yaml]: I don't think this is exposed in any release yet. @PiotrSikora @jplevyak to double check.

**Review by rshriram — APPROVED** [review]

**Review by geeknoid — APPROVED** [review]

**Review by douglas-reid — APPROVED** [review]

**Pull request merged**